src/libs/Fullstory/index.native.ts

```diff
@@ -63,5 +70,42 @@ const FS = {
     },
 };
 
+/**
+ * Placeholder function for Mobile-Web compatibility.
+ */
+function parseFSAttributes(): void {
+    // pass
+}
+
+/*
+    prefix? if component name should be used as a prefix,
+    in case data-test-id attribute usage,
+    clean component name should be preserved in data-test-id.
+*/
+function getFSAttributes(name: string, mask: boolean, prefix: boolean): string {
+    if (!name && !prefix) {
+        return `${mask ? MASK : UNMASK}`;
+    }
+    // prefixed for Native apps should contain only component name
+    if (prefix) {
+        return name;
+    }
+
+    return `${name},${mask ? MASK : UNMASK}`;
+}
+
+function getChatFSAttributes(context: OnyxEntry<PersonalDetailsList>, name: string, report: OnyxInputOrEntry<Report>): string {
```

Comment:
did some benchmark and it is better to return earlier, also I think is cleaner
<img width="954" alt="Screenshot 2024-11-07 at 21 14 14" src="https://github.com/user-attachments/assets/110d4c15-9230-4dd6-bcdf-adafbfc8ba4b">

```suggestion
function getChatFSAttributes(context: OnyxEntry<PersonalDetailsList>, name: string, report: OnyxInputOrEntry<Report>): string[] {
    if (!name) {
        return '';
    }

    if (isConciergeChatReport(report)) {
        const formattedName = `${CONCIERGE}-${name}`;
        return [`${formattedName}`, `${UNMASK},${formattedName}`];
    } 

    if (isExpensifyAndCustomerChat(context, report)) {
        const formattedName = `${CUSTOMER}-${name}`;
        return [`${formattedName}`, `${UNMASK},${formattedName}`];
    }

    const formattedName = `${OTHER}-${name}`;
    return [`${formattedName}`, `${MASK},${formattedName}`];
}
```